libtock/ros.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "tock.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define ROS_DRIVER_NUM 0x00009
+
+// We currently support ROS version 1
+// Version 1:
+//   |-------------------------|
+//   |       Count (u32)       |
+//   |-------------------------|
+//   |   Pending Tasks (u32)   |
+//   |-------------------------|
+//   |                         |
+//   |     Time Ticks (u64)    |
+//   |-------------------------|
+#define ROS_BUFFER_LEN (4 * 4 + 4 * 4 + 8 * 4)
+
+// Get the latest version of the read only state supported by the kernel.
+int ros_get_version(void);
```

Comment:
Should this maybe be some form of `bool read_only_state_is_supported_version(void)` ?

My thinking here is that library users don't really care what the version is, what they care about is whether the library is going to work correctly. This seems like a more future-proof interface to expose to users?

Reply:
I actually think it's the other way around. Library users do care about the version, for example they might want to do this

```c
if (ros_get_version() >= 2) {
    /* Version 2 supports this new function, so let's use that */
    ros_cool_function();
} else {
   /* Let's do it the old fashioned way */
   really_slow_function();
}
```

Reply:
Yeah, that's true as well. My concern is that people can write code that doesn't check the version and then things break out from underneath them, but I suppose that's really a user-error issue, and we can't save them from everything.